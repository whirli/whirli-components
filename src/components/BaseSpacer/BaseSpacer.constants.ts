export const PropKeys = {
  SIZE: 'size',
};

export const PropValues = {
  [PropKeys.SIZE]: ['default'],
};
