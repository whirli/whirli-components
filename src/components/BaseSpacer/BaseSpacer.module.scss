@import '../../styles/Spacers.scss';
@import '../../styles/Breakpoints√ß.scss';

// default styles
.spacer {
    display: block;
    width: 100%;
}

// breakpoint styles
$packageBreakpointStyles: (
    size-0: (
        height: 0
    ),
    size-2xs: (
        height: $spacer2Xs,
    ),
    size-xs: (
        height: $spacerXs,
    ),
    size-sm: (
        height: $spacerSm,
    ),
    size-default: (
        height: $spacerDefault,
    ),
    size-md: (
        height: $spacerMd,
    ),
    size-lg: (
        height: $spacerLg,
    ),
    size-xl: (
        height: $spacerXl,
    ),
    size-2xl: (
        height: $spacer2Xl,
    ),
    size-3xl: (
        height: $spacer3Xl,
    ),
    size-4xl: (
        height: $spacer4Xl,
    ),
    size-5xl: (
        height: $spacer5Xl,
    ),
    size-6xl: (
        height: $spacer6Xl,
    ),
    size-7xl: (
        height: $spacer7Xl,
    ),
);