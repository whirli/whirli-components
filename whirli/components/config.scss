@import '@whirli-components/styles/Helpers.scss';

$breakpointPrefix: BREAKPOINT_;

$breakpoints: (
    tablet: (
        prefix: t,
        min-width: 672px,
    ),
    small_desk: (
        prefix: sd,
        min-width: 1056px
    ),
    large_desk: (
        prefix: ld,
        min-width: 1312px
    ),
    huge_desk: (
        prefix: hd,
        min-width: 1584px
    ),
);

:export {
    prefix: $breakpointPrefix;
    #{$breakpointPrefix}default: 0px;
    @each $name, $value in $breakpoints {
        #{$breakpointPrefix}#{$name}: map-deep-get($breakpoints, #{$name}, 'min-width')
    }
}