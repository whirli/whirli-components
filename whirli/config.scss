@import '@whirli-components/src/styles/Helpers.scss';

$breakpoints: (
    tablet: (
        prefix: t,
        min-width: 48rem,
    ),
    desktop: (
        prefix: d,
        min-width: 80rem,
    )
);

:export {
    @each $name, $value in $breakpoints {
        BREAKPOINT_#{to-upper-case(#{$name})}: map-deep-get($breakpoints, #{$name}, 'min-width')
    }
}