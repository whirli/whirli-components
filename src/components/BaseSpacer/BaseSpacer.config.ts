// @ts-ignore
import { PropType } from 'vue';
import * as PropHelpers from '@whirli-components/helpers/props';
import { ComponentStyles, ComponentProps } from '@whirli-components/@types/components';

// Local config
import config from '@whirli-local/components/BaseSpacer/BaseSpacer.config';

// Package config
import { PropKeys } from './BaseSpacer.constants';
import type { PropSize } from './BaseSpacer.types';

export const ConfigStyles: ComponentStyles = {
  ...config.styles,
  [PropKeys.SIZE]: {
    generateBreakpoints: true,
    classes: {
      '0': 'size-0',
      '4xs': 'size-4xs',
      '3xs': 'size-3xs',
      '2xs': 'size-2xs',
      xs: 'size-xs',
      sm: 'size-sm',
      default: 'size-default',
      md: 'size-md',
      lg: 'size-lg',
      xl: 'size-xl',
      '2xl': 'size-2xl',
      '3xl': 'size-3xl',
      '4xl': 'size-4xl',
      '5xl': 'size-5xl',
      ...config.styles[PropKeys.SIZE]?.classes,
    } as Record<PropSize, string>,
  },
};

export const ConfigProps: ComponentProps = {
  [PropKeys.SIZE]: {
    required: false,
    type: [String, Object] as PropType<PropSize>,
    default: 'default',
    ...PropHelpers.breakpointStylePropValidation(ConfigStyles, PropKeys.SIZE),
  },
  ...config.props,
};
