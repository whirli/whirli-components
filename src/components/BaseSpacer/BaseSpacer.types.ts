import { PropKeys, PropValues } from './BaseSpacer.constants';

const propSize = PropValues[PropKeys.SIZE];

export type PropSize = typeof propSize[number];
