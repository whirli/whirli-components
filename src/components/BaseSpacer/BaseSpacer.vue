<template>
  <div :class="classes" />
</template>

<script setup="props" lang="ts">
// Vue
import { defineProps } from 'vue';

// Styles
import styles from '@whirli/BaseSpacer/BaseSpacer.module.scss';

// Data
import { ConfigStyles, ConfigProps } from './BaseSpacer.config';

const ComponentStyles = ConfigStyles;

const props = defineProps(ConfigProps);

// Classes
import useClasses from '../../@use/class';
const { makeClasses } = useClasses();
const classes = [styles.spacer, ...makeClasses(ComponentStyles, props, styles)];
</script>
