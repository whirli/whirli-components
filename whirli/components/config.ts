import BaseAside from './BaseAside/BaseAside.config';
import BaseButton from './BaseButton/BaseButton.config';
import BaseColumn from './BaseColumn/BaseColumn.config';
import BaseContainer from './BaseContainer/BaseContainer.config';
import BaseGrid from './BaseGrid/BaseGrid.config';
import BaseImage from './BaseImage/BaseImage.config';
import BaseMain from './BaseMain/BaseMain.config';
import BaseSection from './BaseSection/BaseSection.config';
import BaseSpacer from './BaseSpacer/BaseSpacer.config';
import BaseText from './BaseText/BaseText.config';

export default {
  BaseAside,
  BaseButton,
  BaseColumn,
  BaseContainer,
  BaseGrid,
  BaseImage,
  BaseMain,
  BaseSection,
  BaseSpacer,
  BaseText,
};
