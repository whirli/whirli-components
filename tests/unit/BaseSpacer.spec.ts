import { shallowMount } from '@vue/test-utils';
import { propReturnsClass, propReturnsBreakpointClasses } from '../helpers';

import BaseSpacer from '@/components/BaseSpacer/BaseSpacer.vue';
import { PropKeys } from '@/components/BaseSpacer/BaseSpacer.constants';
import { ConfigStyles } from '@/components/BaseSpacer/BaseSpacer.config';

let wrapper: any;

describe('BaseSpacer.vue - Single classes', () => {
  beforeEach(() => {
    wrapper = shallowMount(BaseSpacer, {
      props: {
        size: 'md',
      },
    });
  });

  it('renders the component', () => {
    expect(wrapper.element).toMatchSnapshot();
  });

  it('maps to the correct styles', () => {
    expect(propReturnsClass(wrapper, ConfigStyles, PropKeys.SIZE, 'size-md')).toBe(true);
  });
});

describe('BaseSpacer.vue - Breakpoint classes', () => {
  beforeEach(() => {
    wrapper = shallowMount(BaseSpacer, {
      props: {
        size: { default: 'md', tablet: 'lg' },
      },
    });
  });

  it('renders the component', () => {
    expect(wrapper.element).toMatchSnapshot();
  });

  it('maps to the correct styles', () => {
    expect(
      propReturnsBreakpointClasses(wrapper, ConfigStyles, PropKeys.SIZE, {
        default: 'size-md',
        tablet: 'size-lg',
      })
    ).toBe(true);
  });
});
