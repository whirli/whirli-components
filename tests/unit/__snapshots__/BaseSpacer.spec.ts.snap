// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BaseSpacer.vue - Breakpoint classes renders the component 1`] = `
<div
  class=""
/>
`;

exports[`BaseSpacer.vue - Single classes renders the component 1`] = `
<div
  class=""
/>
`;
